api/src/main/java/datawave/webservice/model/FieldMapping.java

```diff
@@ -0,0 +1,51 @@
+package datawave.webservice.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.apache.commons.lang.builder.CompareToBuilder;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
+import java.io.Serializable;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@XmlAccessorType(XmlAccessType.NONE)
+public class FieldMapping implements Serializable, Comparable<FieldMapping> {
+    
+    private static final long serialVersionUID = 1L;
+    
+    @XmlAttribute(required = true)
+    private String datatype;
+    
+    @XmlAttribute(required = true)
+    private String fieldName;
+    
+    @XmlAttribute(required = true)
+    private String modelFieldName;
+    
+    @XmlAttribute(required = true)
+    private Direction direction;
+    
+    @XmlAttribute(required = true)
+    private String columnVisibility;
+    
+    @Override
+    public int compareTo(FieldMapping obj) {
+        
+        if (obj == null) {
+            throw new IllegalArgumentException("can not compare null");
+        }
+        
+        if (obj == this)
+            return 0;
+        
+        return new CompareToBuilder().append(datatype, ((FieldMapping) obj).datatype).append(fieldName, ((FieldMapping) obj).fieldName)
```

Comment:
Seems odds to have to cast inside each append call. Maybe this way is cleaner nesting the CompareToBuilder inside an `instanceOf` check.

Reply:
Agreed.  This isn't new code - it was moved from here: https://github.com/NationalSecurityAgency/datawave/blob/integration/web-services/client/src/main/java/datawave/webservice/model/FieldMapping.java

Seems we've been doing it this way for some time.  I'm removing the casts in the spring-boot-2.x-accumulo-2.1 branch